mono/utils/mono-mmap.c

```diff
@@ -351,7 +351,8 @@ mono_mprotect (void *addr, size_t length, int flags)
 			memset (addr, 0, length);
 #else
 		memset (addr, 0, length);
-#ifdef HAVE_MADVISE
+/* AIX doesn't have MADV_FREE */
+#if defined(HAVE_MADVISE) && !defined(_AIX)
```

Comment:
why isn't `HAVE_MADVISE` detected properly by autotools?

Reply:
We do have that function, but AIX doesn't support the MADV_FREE flag.

Reply:
is it a define (or enum or const int) on all/most systems? i.e. can you ifdef the flag?